server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -740,7 +740,9 @@ protected void doRun() {
                         totalMetrics.preIngest();
                         final int slot = i;
                         final Releasable ref = refs.acquire();
-                        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver();
+                        final DocumentSizeObserver documentSizeObserver = documentParsingProvider.newDocumentSizeObserver(
+                            indexRequest.index()
```

Comment:
Ingest processors can change the index name, right? So then do we need this to be lazily set? Perhaps a method on the observer itself?

Reply:
I can make this lazy - set after ingestpipelines, but I suspect this is rare and would mean we are executing metering logic when we don't need it?
Requests to system indices cannot originate from regular users, and also regular users should not be able to reroute their indexing requests to system indices. Although I cannot see any validation for this..